### PR TITLE
fix correct package name setup py

### PR DESCRIPTION
**Pull request by sondrewb**
Resolves (partially, some changes needed in poetry too): https://github.com/python-poetry/poetry/issues/2354

- [x] Added **tests** for changed code.
- [x] Updated **documentation** for changed code.

### CODE REVIEW

**Comment by abn** [comment]
@sondrewb can you add positive and negative test cases for this please?

**Comment by sondrewb** [comment]
> @sondrewb can you add positive and negative test cases for this please?

Sure can do! Should've done that in the first place. Give me some days and I will try to add some.

**Comment by sondrewb** [comment]
@abn I added some tests that should cover the issue trying to be solved. WDYT?

**Comment by sondrewb** [comment]
@finswimmer

File: poetry/core/masonry/utils/include.py [hunk 1]
```diff
@@ -27,6 +27,10 @@ def __init__(
 
         self._elements = sorted(list(self._base.glob(str(self._include))))
 
+    @property
+    def include(self):  # type: () -> Path
+        return self._include
+
```
Comment (sdispater) [inline poetry/core/masonry/utils/include.py:33]: Unless I am missing something, this is not strictly needed for this change, right? It's possible to just use `self._include` below.

File: poetry/core/masonry/utils/package_include.py [hunk 2]
```diff
@@ -66,7 +66,7 @@ def check_elements(self):  # type: () -> PackageInclude
         else:
             if root.is_dir():
                 # If it's a directory, we include everything inside it
-                self._package = root.name
+                self._package = self.include.replace("/", ".")
```
Comment (sdispater) [inline poetry/core/masonry/utils/package_include.py:69]:
```suggestion
                self._package = self._include.replace("/", ".")
```

File: poetry/core/masonry/utils/include.py (hunk 1 above)
Comment (sondrewb) [reply poetry/core/masonry/utils/include.py:33]: Indeed. I added the @property because it seemed to be the pattern in other parts of the code.

**Comment by sondrewb** [comment]
@sdispater I believe I have made the requested changes now.

File: poetry/core/masonry/utils/package_include.py [hunk 3]
```diff
@@ -66,7 +66,7 @@ def check_elements(self):  # type: () -> PackageInclude
         else:
             if root.is_dir():
                 # If it's a directory, we include everything inside it
-                self._package = root.name
+                self._package = self._include.replace("/", ".")
```
Comment (abn) [inline poetry/core/masonry/utils/package_include.py:69]: @sondrewb one last question; is platform specific path seperators an issue here?

Comment (abn) [reply poetry/core/masonry/utils/package_include.py:69]: Any updates here?

Comment (david-fortini) [reply poetry/core/masonry/utils/package_include.py:69]: @abn If it is an idea, I can work on this and finish it off next week if @sondrewb does not have time for it. let me know

Comment (abn) [reply poetry/core/masonry/utils/package_include.py:69]: Happy to entertain a new pr. Maintaining the current change attributions of course. :)

Comment (david-fortini) [reply poetry/core/masonry/utils/package_include.py:69]: Of course :100:  Should I mention @sondrewb on the PR or you meant that I fork his project instead?

Comment (abn) [reply poetry/core/masonry/utils/package_include.py:69]: Neither. Just rebase their commits onto your fork/branch. Simply retain the original commit author, if squashing use [`Co-Authored-By`](https://docs.github.com/en/github/committing-changes-to-your-project/creating-a-commit-with-multiple-authors).

**Pull request closed without merging**